web/webkit/src/main/scala/net/liftweb/http/LiftRules.scala

```diff
@@ -2295,3 +2301,80 @@ trait FormVendor {
   private object requestForms extends SessionVar[Map[String, List[FormBuilderLocator[_]]]](Map())
 }
 
+abstract class LiftRulesSettingException(settingName: String, stackTrace: LiftRules.StackTrace, message: String) extends Exception(message) {
+  setStackTrace(stackTrace)
+}
+
+case class LiftRulesSettingWrittenAfterReadException(settingName: String, stackTrace: LiftRules.StackTrace, message: String)
+  extends LiftRulesSettingException(settingName, stackTrace, message)
+
+case class LiftRulesSettingWrittenAfterBootException(settingName: String, stackTrace: LiftRules.StackTrace, message: String)
+  extends LiftRulesSettingException(settingName, stackTrace, message)
+
+case class LiftRulesSettingWrittenTwiceException(settingName: String, stackTrace: LiftRules.StackTrace, message: String)
+  extends LiftRulesSettingException(settingName, stackTrace, message)
+
+
+class LiftRulesSetting[T](val name: String, val default: T) extends LiftValue[T] with HasCalcDefaultValue[T] {
+  import LiftRules.StackTrace
+
+  private [this] var v: T = default
+  private [this] var lastSet: Option[Exception] = None
+  private [this] var lastRead: Option[Exception] = None
+
+  private [this] def writeAfterReadMessage =
+    s"LiftRules.$name was set AFTER already being read! " +
+    s"Your Lift application may be booting into an inconsistent state. " +
+    s"Review the stacktrace below to see where the value was last read. " +
+    s"Consider re-ordering your settings in boot. "
+
+  private [this] def writeAfterBootMessage =
+    s"LiftRules.$name set after Lift finished booting. " +
+    s"This could potentially cause your Lift application to run in an inconsistent state. " +
+    s"Review the stacktrace below to see where settings are being changed after boot time. "
+
+  private [this] def writtenTwiceMessage1(newVal: T) =
+    s"LiftRules.$name was set to $v then later set to $newVal. " +
+    s"This could potentially cause your Lift application to run in an inconsistent state. " +
+    s"Review the stacktrace below to see where LiftRules.$name was first set to $v. "
+
+  private [this] def writtenTwiceMessage2(newVal: T) =
+    s"Review the stacktrace below to see where LiftRules.$name was later set to $newVal. "
+
+  private [this] val toIgnore = Set("LiftRulesSetting", "LiftValue")
+  private [this] def trimmedStackTrace(t: Throwable): StackTrace =
+    t.getStackTrace.dropWhile(e => toIgnore.find(e.getClassName contains _).isDefined)
```

Comment:
We can look up FQNs at runtime though, right? Just `Set(LiftRulesSetting.getClass.getName, LiftValue.getClass.getName)` or somesuch? If not, I'm down with this---it's not a huge deal anyway.